### PR TITLE
Updated Settings Activity to new MVP design

### PR DESCRIPTION
**Pull request by mayank408**
Fixes issue #841

Updated Settings Activity to new MVP design

 I have not done model part that will done by @amitiwary999 So its blank for now.

### CODE REVIEW

File: app/src/main/java/org/fossasia/susi/ai/data/LoginModel.kt [hunk 1]
```diff
@@ -14,6 +14,7 @@ import retrofit2.Response
  *
  * Created by chiragw15 on 4/7/17.
  */
+
```
Comment (amitiwary999) [inline app/src/main/java/org/fossasia/susi/ai/data/LoginModel.kt]: remove extra space.

File: app/src/main/java/org/fossasia/susi/ai/data/UtilModel.kt [hunk 2]
```diff
@@ -14,6 +19,7 @@ import retrofit2.Response
  *
  * Created by chiragw15 on 10/7/17.
  */
+
```
Comment (amitiwary999) [inline app/src/main/java/org/fossasia/susi/ai/data/UtilModel.kt]: remove extra space here.

File: app/src/main/java/org/fossasia/susi/ai/settings/SettingsActivity.kt [hunk 3]
```diff
@@ -22,7 +27,6 @@ class SettingsActivity : AppCompatActivity(), ISettingsView {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-
```
Comment (amitiwary999) [inline app/src/main/java/org/fossasia/susi/ai/settings/SettingsActivity.kt]: no need to remove this space.

File: app/src/main/java/org/fossasia/susi/ai/data/contract/ISettingsModel.kt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package org.fossasia.susi.ai.data.contract
+
+/**
+ * The interface for Settings Model
+ *
+ * Created by mayanktripathi on 07/07/17.
+ */
+
+interface ISettingsModel
```
Comment (amitiwary999) [inline app/src/main/java/org/fossasia/susi/ai/data/contract/ISettingsModel.kt]: as you mentioned that I will work on model part. So remove it I will add it.

File: app/src/main/java/org/fossasia/susi/ai/data/SettingsModel.kt [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+/**
+ * Model of Settings
+ * The M in MVP
+ * Stores all business logic
+ *
+ * Created by mayanktripathi on 07/07/17.
+ */
+
+class SettingsModel: ISettingsModel
```
Comment (amitiwary999) [inline app/src/main/java/org/fossasia/susi/ai/data/SettingsModel.kt]: as you mentioned I will work on model part. So remove it I will work on it.

File: app/src/main/java/org/fossasia/susi/ai/activities/MainActivity.java [hunk 6]
```diff
@@ -981,7 +980,7 @@ private void checkPermissions() {
 
         if(ActivityCompat.checkSelfPermission(this,
                 Manifest.permission.RECORD_AUDIO) == PackageManager.PERMISSION_GRANTED) {
-            PrefManager.putBoolean(Constant.MIC_INPUT, checkMicInput());
+            PrefManager.putBoolean(Constant.MIC_INPUT, checkMicInput() && PrefManager.getBoolean(Constant.MIC_INPUT, false));
```
Comment (chiragw15) [inline app/src/main/java/org/fossasia/susi/ai/activities/MainActivity.java]: Avoid changes in the main activity, please. The main activity will be removed at all and these changes won't reflect then. Instead, comment these changes with the line number in #832. I'll add them.

Comment (mayank408) [inline app/src/main/java/org/fossasia/susi/ai/activities/MainActivity.java]: @amitiwary999 Pls do this change while adding the feature sending the settings to server. 
@chiragw15 I have added the line comment.

**Comment by mariobehling** [comment]
Circle CI failing.

**Pull request merged**